bdk_electrum_example/src/electrum.rs

```diff
@@ -0,0 +1,236 @@
+use std::{collections::BTreeMap, ops::Deref};
+
+use bdk_cli::Broadcast;
+use bdk_core::{
+    bitcoin::{BlockHash, Script, Txid},
+    sparse_chain::{InsertCheckpointErr, InsertTxErr, SparseChain, UpdateFailure},
+    BlockId, TxHeight,
+};
+use electrum_client::{Client, Config, ElectrumApi};
+
+#[derive(Debug)]
+pub enum ElectrumError {
+    Client(electrum_client::Error),
+    InsertTx(InsertTxErr),
+    Update(UpdateFailure),
+    InsertCheckpoint(InsertCheckpointErr),
```

Comment:
Each of these errors needs docs. It should be clear how a user is meant to recover from each of them. If I get a client error I guess I know that something went wrong connecting to the electrum client but with the others I have no clue what it means.

Reply:
Removed these error variants in latest push..